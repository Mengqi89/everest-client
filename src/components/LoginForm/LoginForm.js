import React, { Component } from 'react'
import { withRouter } from 'react-router-dom'
import TokenService from '../../services/token-service'
import SchoolApiService from '../../services/school-api-service'
import AdminApiService from '../../services/admin-api-service'
import TeacherApiService from '../../services/teacher-api-service'

import './LoginForm.scss'

class LoginForm extends Component {
  state = {
    accountType: null,
    error: null
  }
  handleAccountTypeChange = (ev) => {
    this.setState({
      accountType: ev.target.value
    })
  }

  handleSubmitJwtAuth = ev => {
    ev.preventDefault()
    this.setState({ error: null })
    const { username, password } = ev.target
    if (this.state.accountType === 'school') {
      SchoolApiService.postLogin({
        username: username.value,
        password: password.value
      })
        .then(res => {
          username.value = ''
          password.value = ''

          TokenService.saveAuthToken(res.authToken)
          TokenService.saveUserType('school')
          this.props.history.push('/profile')
          // this.context.handleLoginSucces()
        })
        .catch(res => {
          this.setState({ error: res.error })
        })
    }
    if (this.state.accountType === 'admin') {
      AdminApiService.postLogin({
        username: username.value,
        password: password.value
      })
        .then(res => {
          username.value = ''
          password.value = ''

          TokenService.saveAuthToken(res.authToken)
          TokenService.saveUserType('admin')
          this.props.onLoginSuccess()
        })
        .catch(res => {
          this.setState({ error: res.error })
        })
    }
    if (this.state.accountType === 'teacher') {
      TeacherApiService.postLogin({
        username: username.value,
        password: password.value
      })
        .then(res => {
          username.value = ''
          password.value = ''

          TokenService.saveAuthToken(res.authToken)
          TokenService.saveUserType('teacher')
          this.props.onLoginSuccess()
        })
        .catch(res => {
          this.setState({ error: res.error })
        })
    }
  }

  render() {
    const { error } = this.state

    return (
      <form className="LoginForm" onSubmit={this.handleSubmitJwtAuth}>
        <div role="alert">{error && <p className="red">{error}</p>}</div>
        <fieldset>
          <legend htmlFor="account-type">I am a</legend>
          <label htmlFor="teacher">Teacher</label>
          <input type="radio" id="teacher" name="account-type" value="teacher" onChange={this.handleAccountTypeChange} />
          <label htmlFor="teacher">School</label>
          <input type="radio" id="school" name="account-type" value="school" onChange={this.handleAccountTypeChange} />
          <label htmlFor="teacher">Admin</label>
          <input type="radio" id="admin" name="account-type" value="admin" onChange={this.handleAccountTypeChange} />
        </fieldset>
        <div className="username">
          <label htmlFor="username">Username</label>
          <input required name="username" id="username" type="username" autoComplete="new-password" />
        </div>
        <div className="password">
          <label htmlFor="password">Password</label>
          <input
            autoComplete="new-password"
            required
            name="password"
            type="password"
            id="password"
          />
        </div>
        <button type="submit">Login</button>
      </form >
    );
  }
}

export default withRouter(LoginForm)