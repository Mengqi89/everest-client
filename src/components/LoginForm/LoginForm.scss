@import '../../assets/variables.scss';

.LoginForm {

  legend {
    margin-bottom: .5em;
    color: $accent-color;
  }

  input[type=radio] {
    margin: 0 0.5em;
    width: initial;
    height: initial
  }
  fieldset {
    padding: 0 2em;
    margin: 1em auto;
  }
  input {
    border: none;
    border-bottom: 2px solid $accent-color;
    height: 2em;
    font-size: 1em;
    font-family: $fancy-font;
    &:focus {
      outline-width: 0;
    }
  }

  .username, .password {
    label {
      margin-top: 2em;
      padding: 0 2em;
      color: grey;
    }
    
    input {
      margin: 0 2em;
    }

    display: flex;
    flex-direction: column;
  }
  button {
    @include green-button;
    width: 82px;
    height: 34px;
   display: flex;
   align-items: center;
   justify-content: center;
   margin-bottom: 1.5em;

  }
}