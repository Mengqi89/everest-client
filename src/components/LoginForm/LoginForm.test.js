import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import LoginForm from './LoginForm';

describe('LoginForm route component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <LoginForm />
      </BrowserRouter>
      , div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <LoginForm />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})