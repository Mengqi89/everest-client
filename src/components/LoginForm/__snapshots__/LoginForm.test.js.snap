// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm route component renders the UI as expected 1`] = `
<form
  className="LoginForm"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div
    className="username"
  >
    <label
      htmlFor="username"
    >
      Username
    </label>
    <input
      autoComplete="new-password"
      id="username"
      name="username"
      required={true}
      type="username"
    />
  </div>
  <div
    className="password"
  >
    <label
      htmlFor="password"
    >
      Password
    </label>
    <input
      autoComplete="new-password"
      id="password"
      name="password"
      required={true}
      type="password"
    />
  </div>
  <button
    type="submit"
  >
    Login
  </button>
  <fieldset
    id="admin-login"
  >
    <input
      id="admin-input"
      name="account-type"
      onClick={[Function]}
      type="checkbox"
      value="admin"
    />
    <label
      htmlFor="admin-input"
      id="admin-label"
    >
      Check this only if you are an administrator.
    </label>
  </fieldset>
</form>
`;
